widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java

```diff
@@ -208,6 +213,12 @@ public static void setControllerFactory(ControllerFactory controllerFactory) {
         Dependencies.controllerFactory = controllerFactory;
     }
 
+    public static Navigator getNavigator(@Nullable ArrayList<String> queues, @Nullable String contextAssetId) {
+        navigator.setQueueIds(queues);
+        navigator.setContextAssetId(contextAssetId);
+        return navigator;
```

Comment:
What are the benefits of having `navigator` as a singleton?
Maybe it is better to create an instance for each request and take queues and contextId as a constructor argument.

We still can use a singleton repository to deal with queues if needed.

Reply:
Good point, I also thought about it. Changed 5867be8217773a720d124d5409802717a6712c37